core/domain/exp_domain.py

```diff
@@ -1658,7 +1658,8 @@ def validate(self, strict: bool = False) -> None:
             state.validate(
```

Comment:
Adding a note to self, Add backend validation for the below scenario:
Need to check if the HasElementXAtPositionY inside the DragAndDrop interaction has valid position. Need to be applied for the strict exps. Consider the scenario - Equals rule having all the options at position 1 and we have HasElementXAtPositionY rule having Y as value 3. Now the HasElementXAtPositionY is invalid and it will never going to match.

Reply:
Done!